### DISCUSSION TITLE
Theme III/8 PF, feature type "ProductionFacility", attribute "function", problematic testing of NACE code URL

### DISCUSSION THREAD

**Discussion by fantamir** [INSPIRE-MIF/helpdesk-validator#1252] [Ideas]
Dear colleagues,

The attribute **function** in the feature type **ProductionFacility** of the INSPIRE theme **III/8 Production and Industrial Facilities** should contain URL to relevant **NACE code** as the preferred choice for European interoperability.
**example:**
_<act-core:function>
        <act-core:Function>
          <act-core:activity xlink:href="https://inspire.ec.europa.eu/codelist/EconomicActivityNACEValue/C.23.51"></act-core:activity>
        </act-core:Function>
</act-core:function>_
This example is accepted by latest version of INSPIRE validator (implemented at 16th of September) without any problems.

In the dataset I am preparing for INSPIRE purposes, a NACE code is unpopulated for some of spatial objects. Unfortunately, the NACE codelist does not contain any item having value _Unpopulated_,  _Unknown_ or something like that. Therefore, the reasonable solution could be to fill the **nilReason** parameter of the **function.Function.activity** attribute in such case, instead of **href**.
**example:**
_<act-core:activity nilReason="http://inspire.ec.europa.eu/codelist/VoidReasonValue/Unpopulated"></act-core:activity>_
Unfortunately, this solution is rejected by INSPIRE validator.

On the other hand, all four examples below are accepted by validator without any problems!!:
_<act-core:activity xlink:href="https://inspire.ec.europa.eu/codelist/EconomicActivityNACEValue/"></act-core:activity>
<act-core:activity xlink:href="http://inspire.ec.europa.eu/codelist/VoidReasonValue/Unpopulated"></act-core:activity>
<act-core:activity xlink:href="valueUnpopulated"></act-core:activity>
<act-core:activity xlink:href=""></act-core:activity>_

It seems, the validator just tests the presence of href parameter, without checking of its content. This is very strange.

**Suggested solution:**
- add testing of the content of **href** parameter of the **function.Function.activity** attribute; allow the URLs directed to the items of NACE (or other similar) codelist only; if the set of such codelists cannot be compiled, then the test if a href parameter contains a valid URL should be implemented at least.
- allow using of **nilReason** parameter in the **function.Function.activity** attribute
  or 
  add a new item containing the value **Unpopulated** to the NACE codelist, to create the URL _https://inspire.ec.europa.eu/codelist/EconomicActivityNACEValue/Unpopulated_ , which could be used in the **href** parameter
- implement the solution described above on all cases in the INSPIRE themes, where NACE (or other similar) code should be filled by data providers.

All spatial objects with missing NACE code value should be deleted from the INSPIRE datasets, because function.Function.activity is a mandatory attribute. Since such deleting would reduce the usability of our dataset, I intend to use _<act-core:activity xlink:href="http://inspire.ec.europa.eu/codelist/VoidReasonValue/Unpopulated"></act-core:activity>_ solution for the time being.

Miroslav Fanta
CENIA, Czech Environmental Information Agency

**Comment by fabiovin** [comment]
Dear @fantamir,

I will further analyse the checks carried out by the validator, in the meantime, I can anticipate to you that the _function.Function.activity_ attribute does not have the stereotype "voidable", so it cannot be void. 

For this reason, the _nilReason_ attribute should not be present inside the _function.Function.activity_ element.

Could you please provide a sample GML file you are using for the test?

![image](https://user-images.githubusercontent.com/46569952/133819443-d9d2f03e-32c2-4e63-acf2-d419c2acf0b3.png)

**Comment by fantamir** [comment]
Dear Fabio,

thank you very much for your prompt feedback.
Sample GML file is attached. Focus of rows containing comments  - example 2, ..5
[IPPC_ETRS89_NACE_sample1_GML_GitHub609.zip](https://github.com/INSPIRE-MIF/helpdesk-validator/files/7192968/IPPC_ETRS89_NACE_sample1_GML_GitHub609.zip)

INSPIRE validator test report is here:
https://inspire.ec.europa.eu/validator/test-run/details.html?id=EID768c7b2a-16fb-4901-b74d-756b5056e5e7
(skipped test: 0, failed test: 0, warnings: 0)

Best wishes
Miroslav

**Comment by fabiovinci** [comment]
Dear all,
since this issue requires the fixing of the related ETS that is currently unplanned, it will be moved to the related discussion section.
